python/cuxfilter/charts/deckgl/plots.py

```diff
@@ -48,6 +50,31 @@ class Choropleth(BaseChoropleth):
         "controller": True,
     }
 
+    def compute_colors(self):
+        if self.geo_color_palette is None:
+            self.geo_color_palette = bokeh.palettes.Purples9
+
+        self.source_df = self.source.to_df()
+        BREAKS = np.linspace(
+            self.source_df[self.color_column].min(),
+            self.source_df[self.color_column].max(),
+            len(self.geo_color_palette),
+        )
+
+        def color_scale(val):
+            if np.isnan(val):
+                return list(ImageColor.getrgb(self.nan_color)) + [50]
+            for i, b in enumerate(BREAKS):
+                if val < b:
+                    return list(
+                        ImageColor.getrgb(self.geo_color_palette[i])
+                    ) + [255]
+            return list(ImageColor.getrgb(self.geo_color_palette[i])) + [255]
+
+        self.source_df[self.rgba_columns] = pd.DataFrame(
+            self.source_df[self.color_column].apply(color_scale).tolist()
+        )
```

Comment:
I feel like you [`pandas.cut`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.cut.html) could be used here to efficiently get the palette indices and then the colors could be computed with a list comprehension (untested sketch)
```
nan_color = list(ImageColor.getrgb(self.nan_color)) + [50]

x = self.source_df[self.color_column]
inds = pd.cut(x, BREAKS, labels=False, include_lowest=True)

colors = [nan_color if np.isnan(i) else list(ImageColor.getrgb(self.geo_color_palette[i])) + [255] for i in inds]
```

Reply:
Yes, the proposed method is actually around 5-6x faster (1000 rows)